File: backend/app/models/TagValue.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from datetime import datetime
from sqlalchemy.orm import relationship
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class TagValue(Base):
    __tablename__ = 'tag_values'

    tag_value_id = Column(Integer, primary_key=True, autoincrement=True)
    tag_id = Column(Integer, ForeignKey('tags.tag_id'), nullable=False)
    value = Column(String(255), nullable=False)
    created_by = Column(Integer, ForeignKey('admins.admin_id'), nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)

    tag = relationship("Tag", back_populates="tag_values")
    created_by_admin = relationship("Admin", back_populates="tag_values")

    def __repr__(self):
        return f"<TagValue(tag_value_id={self.tag_value_id}, value={self.value})>"

File: backend/TagFlowApi/Models/TagValue.cs
// TagFlowApi/Models/TagValue.cs
namespace TagFlowApi.Models
{
    public class TagValue
    {
        public int TagValueId { get; set; }
        public string Value { get; set; } = "";
        public DateTime CreatedAt { get; set; }

        // Foreign key to Tag table
        public int TagId { get; set; }
        public Tag Tag { get; set; } = null!;

        // Foreign key to Admin table (creator of the tag value)
        public int CreatedBy { get; set; }
        public Admin? CreatedByAdmin { get; set; }

        // One-to-many relationship with file rows
        public ICollection<FileRow> FileRows { get; set; } = new List<FileRow>();
    }
}
